exotations/task_maps/task_map/src/JointVelocityBackwardDifference.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+ *      Author: Christopher E. Mower
+ *
+ * Copyright (c) 2018, University of Edinburgh
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *  * Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *  * Neither the name of  nor the names of its contributors may be used to
+ *    endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+#include "task_map/JointVelocityBackwardDifference.h"
+
+REGISTER_TASKMAP_TYPE("JointVelocityBackwardDifference", exotica::JointVelocityBackwardDifference);
+
+namespace exotica
+{
+constexpr double backward_difference_params_ = -1.0;  //!< Binomial cooeficient parameters.
+
+JointVelocityBackwardDifference::JointVelocityBackwardDifference() = default;
+JointVelocityBackwardDifference::~JointVelocityBackwardDifference() = default;
+
+void JointVelocityBackwardDifference::assignScene(Scene_ptr scene)
+{
+    scene_ = scene;
+
+    // Get ndof
+    N_ = scene_->getSolver().getNumControlledJoints();
+
+    // Ensure start state is correct size
+    init_.StartState.resize(N_, 1);
```

Comment:
This is _not_ a good solution. You are hereby changing _what the user passed in_ internally. It should throw and tell the user the input is wrong. So

```cpp
q_.resize(N_,1);
if (init_.StartState.rows() == 0) q_.setZero(N_);
else if (init_.StartState.rows() == N_) q_ = init_.StartState;
else throw_pretty("..");
```